### PR TITLE
support scalar output in lm

### PR DESCRIPTION
**Pull request by amir-saadat**
this is just a simple update to the code to ensure that scalar function output are supported. Although less of interest, might be considered an edge case.

### CODE REVIEW

**Comment by vroulet** [comment]
Hello Amir!
Thanks for that! Did you check that the current code did not pass your test?

**Comment by vroulet** [comment]
Ok I checked. Indeed the current code does not pass the test.
In the future, it may be better not to use operators such as `@` and keep jax/numpy operators, or, even better, make sure that the code can work with pytrees.

File: jaxopt/_src/levenberg_marquardt.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def init_state(self, init_params: Any, *args,
         increase_factor=2,
         error=tree_l2_norm(gradient),
         residual=residual,
-        loss=0.5 * (residual @ residual),
+        loss=0.5 * jnp.sum(jnp.square(residual)),
```
Comment (vroulet) [inline jaxopt/_src/levenberg_marquardt.py:231]: Since you are at it, loss -> val. Same at other places in the code.

File: tests/levenberg_marquardt_test.py [hunk 2]
```diff
@@ -316,6 +315,14 @@ def test_has_aux_true(self):
     self.assertAllClose(x_opt, x_gt, atol=1e-6)
     self.assertAllClose(state.aux, x_gt**2, atol=1e-6)
 
+  def test_scalar_output_fun(self):
+    lm = LevenbergMarquardt(
+        residual_fun=lambda x: x @ x,
+        tol=1e-6,)
```
Comment (vroulet) [inline tests/levenberg_marquardt_test.py:321]: tol=1e-1

File: tests/levenberg_marquardt_test.py [hunk 3]
```diff
@@ -316,6 +315,14 @@ def test_has_aux_true(self):
...
+    x_init = jnp.ones((10,))
```
Comment (vroulet) [inline tests/levenberg_marquardt_test.py:322]: jnp.ones((2,))

File: tests/levenberg_marquardt_test.py [hunk 4]
```diff
@@ -316,6 +315,14 @@ def test_has_aux_true(self):
...
+    x_opt, _ = lm.run(x_init)
+
+    self.assertAllClose(x_opt, jnp.zeros((10,)), atol=1e-2)
```
Comment (vroulet) [inline tests/levenberg_marquardt_test.py:325]: jnp.zeros((2,)) 
atol=1e0

**Review by vroulet — APPROVED** [review]
